be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -532,6 +532,14 @@ std::vector<OrcPredicate> value_range_to_predicate(
         std::vector<orc::TypeKind>* unsupported_pushdown_types) {
     std::vector<OrcPredicate> predicates;
 
+    PrimitiveType src_type = OrcReader::convert_to_doris_type(type).type;
+    if (src_type != primitive_type) {
+        if (!(is_string_type(src_type) && is_string_type(primitive_type))) {
```

Comment:
why?

Reply:
Altering column type from `string` to `varchar` can still use push-down predicates.